### ISSUE TITLE
Allow to set arbitrary priority for components through REST API

### ISSUE THREAD

**Issue by rob006** [WeblateOrg/weblate#6840]
### Describe the problem

I want to force order of components using priority (by settings different priority for each component).  I tried to use REST API to change these values, but it looks like API accepts only [5 predefined values](https://github.com/WeblateOrg/weblate/blob/ebc31818eea4e82efa4e588693579d9c60c23dd3/weblate/trans/util.py#L43-L49) (I have over 250 components, so it is not even close). Is there any particular reason for this limitation? Priority is stored in database as INT, which allows to store over 4kkk values.

### Describe the solution you'd like

Allow any integer to be set as priority.

**Comment by rob006** [comment]
> It would probably discard the value when editing the settings in the UI if other value would be present.

That would be fine in my case - I would just not use UI to change these settings. This issue is mostly about too restrictive server-side validation - IMO REST API should not inherit UI-specific limitations.

How hard would be to dynamically define additional case if non-standard value is set as priority? So if I set `123` as priority, frontend will have additional "Custom (123)" option in priority select. If there is no non-standard priority, then this field is not visible. That would be some middle-ground solution - non-standard values still could be set only via REST, but UI would not overwrite these settings.

**Comment by nijel** [comment]
I don't think you would be fine with loosing custom priority upon changing component settings in the UI. So at least this needs to be tested and (more likely) implemented.

And, to be precise, this is not a UI limitation, it is defined at the model level currently:

https://github.com/WeblateOrg/weblate/blob/78620d68cb0228fa32144d61fa40df6a31d2ce48/weblate/trans/models/component.py#L597-L604

**Comment by rob006** [comment]
> I don't think you would be fine with loosing custom priority upon changing component settings in the UI.

I do not plan to use UI. And if I do, I expect that script that uses REST API will fix priority after some time.

> And, to be precise, this is not a UI limitation, it is defined at the model level currently

I know that list of possible values is defined at model level, but it is defined there because of the UI. There is no other reason to limit INT field to only 5 values.
I already tried to change priority directly in database and it worked just fine (except UI).
